feat(language_server): introduce `ServerFormatter`

Sysix opened the pull request:
First try on formatter integration. 
The server will only tell the client, that is supports `formatting` when `fmt.experimental` is `true`.

It will respect the flag in `initialize`, `did_change_configuration` (changing configuration), or `did_change_workspace_folders`.
It will not tell the client, it does not support formatting when the configuration is changed to `false`.
This can be optimized but has no high priority, because the client needs to send the request and the `ServerFormatter` needs to be set. The last requirement is then false.

Make `WorkspaceWorker.option` optional, so we the servers knows when the configuration in `initialize` or `initialized` (with `request_workspace_configuration`). Before the server knew it, by looking at the `WorkspaceWorker.server_linter` instance.

camc314 commented:
### Merge activity

* **Sep 20, 6:04 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
feat(language_server): introduce `ServerFormatter` (#13700)

First try on formatter integration.
The server will only tell the client, that is supports `formatting` when `fmt.experimental` is `true`.

It will respect the flag in `initialize`, `did_change_configuration` (changing configuration), or `did_change_workspace_folders`.
It will not tell the client, it does not support formatting when the configuration is changed to `false`.
This can be optimized but has no high priority, because the client needs to send the request and the `ServerFormatter` needs to be set. The last requirement is then false.

Make `WorkspaceWorker.option` optional, so we the servers knows when the configuration in `initialize` or `initialized` (with `request_workspace_configuration`). Before the server knew it, by looking at the `WorkspaceWorker.server_linter` instance.
